### PR TITLE
[action] simplify address generation

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
noticed this while working on #3910

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/builder.go [hunk 1]
```diff
@@ -16,16 +16,16 @@ import (
 	"github.com/iotexproject/iotex-core/pkg/version"
 )
 
+var (
```
Comment (dustinxie) [inline action/builder.go:19]: `var` is placed in front of `type`

File: action/receipt.go [hunk 2]
```diff
@@ -11,20 +11,11 @@ import (
...
 )
 
-var (
-	// StakingBucketPoolTopic is topic for staking bucket pool
-	StakingBucketPoolTopic = hash.BytesToHash256(address.StakingProtocolAddrHash[:])
-
-	// RewardingPoolTopic is topic for rewarding pool
-	RewardingPoolTopic = hash.BytesToHash256(address.RewardingProtocolAddrHash[:])
-)
-
```
Comment (dustinxie) [inline action/receipt.go:27]: defined but not used at all

File: action/candidate_register.go [hunk 3]
```diff
@@ -380,16 +380,11 @@ func ethAddrToNativeAddr(in interface{}) (address.Address, error) {
...
-	if err != nil {
-		return nil, err
-	}
-	ethAddr := common.BytesToAddress(addr.Bytes())
 	data, err := cr.encodeABIBinary()
 	if err != nil {
 		return nil, err
 	}
-	return types.NewTransaction(cr.Nonce(), ethAddr, big.NewInt(0), cr.GasLimit(), cr.GasPrice(), data), nil
+	return types.NewTransaction(cr.Nonce(), _stakingProtocolEthAddr, big.NewInt(0), cr.GasLimit(), cr.GasPrice(), data), nil
```
Comment (dustinxie) [inline action/candidate_register.go:387]: easier way of generating the address

**Review by millken — APPROVED** [review]

**Review by envestcc — APPROVED** [review]
so, `ioAddr` and `ethAddr` have the same `hash`, but the `String()` are different, right?

**Comment by dustinxie** [comment]
> so, `ioAddr` and `ethAddr` have the same `hash`, but the `String()` are different, right?

yes, that's right, they have the same 20-byte `hash`.  Ether's string is just the hex encoding, while IoTeX's string is the bech32 encoding of the underlying 20-byte `hash`

**Pull request merged**